### PR TITLE
Fixing doxygen comments 

### PR DESCRIPTION
**Pull request by tony-josi-aws**
Description
-----------
This PR updates the doxygen config file to the latest main branch config and fixes the doxygen comments.

**NOTE**: WIP. Do not merge yet.

Test Steps
-----------

`doxygen docs/doxygen/config.doxyfile 2>&1 | tee doxyoutput.txt`

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/FreeRTOS_BitConfig.c [hunk 1]
```diff
@@ -237,7 +237,7 @@ void vBitConfig_write_uc( BitConfig_t * pxConfig,
  * @param[in] pxConfig: The structure containing a copy of the bits.
  * @param[in] ucValue: The byte to be written.
  *
- * @return There is no return value. If the operation has failed,
+ * There is no return value. If the operation has failed,
```
Comment (moninom1) [inline source/FreeRTOS_BitConfig.c:240]: Is @return not needed anymore?

Comment (tony-josi-aws) [reply source/FreeRTOS_BitConfig.c:240]: No, as the function `vBitConfig_write_uc` doesnt return anything we dont need that

**Review by shubnil — APPROVED** [review]

**Pull request merged**